### PR TITLE
Adding  socketio python client

### PR DESCRIPTION
**Pull request by chiranjeevfx**
Adding  socketio python client

Screenshot 2023-05-09 at 4 35 35 PM

### CODE REVIEW

File: python/socketioclient/socketio_client.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+import uuid
+import logzero
+import socketio
+
+logger = logzero.logger
+
+ctx = ssl.create_default_context()
+ctx.set_ciphers('DEFAULT')
+
+URL = 'http://localhost:8080'
```
Comment (nikhilmiskin) [inline python/socketioclient/socketio_client.py]: change the url to https://stream.falconx.io?

File: python/socketioclient/socketio_client.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+API_KEY = '<Your API_KEY>'
+PASSPHRASE = '<Your PASSPHRASE>'
+SECRET_KEY = '<Your PASSPHRASE>'
```
Comment (nikhilmiskin) [inline python/socketioclient/socketio_client.py]: minor: Your SECRET_KEY?

File: python/socketioclient/socketio_client.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+                        required=False,
+                        default="BTC/USD",
+                        help="Comma separated token pairs eg. BTC/USD,ETH/USD")
+    parser.add_argument('--levels',
+                        type=lambda s: [int(x) for x in s.split(",")],
+                        required=False,
+                        default="1,2",
+                        help="Comma separated levels eg. 190,200,400")
+
+    parser.add_argument('--print_every_n_response',
```
Comment (nikhilmiskin) [inline python/socketioclient/socketio_client.py]: lets remove this? Not needed from clients perspective

File: python/socketioclient/socketio_client.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        client_request_id = args[0][0]['client_request_id']
+        previous_response_time = self.previous_response_times[client_request_id]
+        streaming_response_time = response_time - previous_response_time
+        running_avg, num_of_responses = self.running_avg_times.get(client_request_id, (0, 0))
+        running_avg = (running_avg * num_of_responses + streaming_response_time) / (num_of_responses + 1)
+        logger.info("printing arguments ", *args)
+        if num_of_responses % self.print_every_n_response == 0:
+            logger.info(f'client_request_id: {client_request_id}, streaming heart beat: {streaming_response_time}, running average: {running_avg}')
+
+        self.running_avg_times[client_request_id] = (running_avg, num_of_responses + 1)
```
Comment (nikhilmiskin) [inline python/socketioclient/socketio_client.py]: all of these calculations are also not needed, this was done only internally to check the response times, just a simple log should be good

File: python/socketioclient/socketio_client.py [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+    def on_connect(self):
+        logger.info('Server connected.')
+        for subscription_request in self.subscription_requests:
+            self.emit('subscribe', subscription_request, namespace='/streaming')
+            client_request_id = subscription_request['client_request_id']
+            self.previous_response_times[client_request_id] = time.time()
+        logger.info('Finished subscribing.')
+
+    # def on_connect(self):
```
Comment (nikhilmiskin) [inline python/socketioclient/socketio_client.py]: minor: remove comments?

File: python/socketioclient/socketio_client.py (hunk 3 above)
Comment (chiranjeevfx) [reply python/socketioclient/socketio_client.py]: Done!

File: python/socketioclient/socketio_client.py (hunk 4 above)
Comment (chiranjeevfx) [reply python/socketioclient/socketio_client.py]: Done!

File: python/socketioclient/socketio_client.py (hunk 5 above)
Comment (chiranjeevfx) [reply python/socketioclient/socketio_client.py]: Done!

File: python/socketioclient/socketio_client.py (hunk 2 above)
Comment (chiranjeevfx) [reply python/socketioclient/socketio_client.py]: Done!

File: python/socketioclient/socketio_client.py (hunk 1 above)
Comment (chiranjeevfx) [reply python/socketioclient/socketio_client.py]: Done!

File: python/socketioclient/socketio_client.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+        logger.info('Finished subscribing.')
+
+    def on_disconnect(self, *args):
+        logger.info('Server disconnected.' + str(args))
+
+    def on_connect_error(self, *args):
+        logger.info("Cannot connect to the server. Error", args)
+
+    def on_response(self, *args):
+        response_time = time.time()
```
Comment (nikhilmiskin) [inline python/socketioclient/socketio_client.py]: hey, lets not do any of this, just a simple log statement should be enough

File: python/socketioclient/socketio_client.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+        client_request_id = args[0]['client_request_id']
+        logger.info("Client received response: " + str(args))
+        previous_response_time = self.previous_response_times[client_request_id]
+        logger.info(f'Client_request_id: {client_request_id}, time taken for first response {response_time - previous_response_time}')
+        self.previous_response_times[client_request_id] = response_time
+
+    def on_stream(self, *args):
+        response_time = time.time()
```
Comment (nikhilmiskin) [inline python/socketioclient/socketio_client.py]: same here, just a log statement with the args

Comment (chiranjeevfx) [reply python/socketioclient/socketio_client.py]: Removed

File: python/socketioclient/socketio_client.py (hunk 6 above)
Comment (chiranjeevfx) [reply python/socketioclient/socketio_client.py]: Changed.

File: python/socketioclient/socketio_client.py [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+import ssl
+import time
+import uuid
+import logzero
+import socketio
+
+logger = logzero.logger
+
+ctx = ssl.create_default_context()
+ctx.set_ciphers('DEFAULT')
```
Comment (akhil9falconx) [inline python/socketioclient/socketio_client.py:14]: why is this needed?
